itest/lnd_open_channel_test.go

```diff
@@ -768,3 +768,80 @@ func testFundingExpiryBlocksOnPending(ht *lntest.HarnessTest) {
 	chanPoint := lntest.ChanPointFromPendingUpdate(update)
 	ht.CloseChannel(alice, chanPoint)
 }
+
+// testSimpleTaprootChannelActivation ensures that a simple taproot channel is
+// active if the initiator stays disconnected in between channel opening and
+// channel confirmation but reconnects after the channel is confirmed.
+func testSimpleTaprootChannelActivation(ht *lntest.HarnessTest) {
+	// Grab the channel participants.
+	alice, bob := ht.Alice, ht.Bob
+
+	// Restart Alice and Bob to activate simple taproot channel feature.
+	simpleTaprootChanArgs := lntest.NodeArgsForCommitType(
+		lnrpc.CommitmentType_SIMPLE_TAPROOT,
+	)
+	ht.RestartNodeWithExtraArgs(alice, simpleTaprootChanArgs)
+	ht.RestartNodeWithExtraArgs(bob, simpleTaprootChanArgs)
+
+	// Make sure Alice and Bob are connected.
+	ht.EnsureConnected(alice, bob)
+
+	// Create simple taproot channel opening parameters.
+	params := lntest.OpenChannelParams{
+		FundMax:        true,
+		CommitmentType: lnrpc.CommitmentType_SIMPLE_TAPROOT,
+		Private:        true,
+	}
+
+	// Alice opens the channel to Bob.
+	pendingChan := ht.OpenChannelAssertPending(alice, bob, params)
+
+	// We'll create the channel point to be able to close the channel once
+	// our test is done.
+	chanPoint := &lnrpc.ChannelPoint{
+		FundingTxid: &lnrpc.ChannelPoint_FundingTxidBytes{
+			FundingTxidBytes: pendingChan.Txid,
+		},
+		OutputIndex: pendingChan.OutputIndex,
+	}
+
+	// We disconnect Bob from Alice before the channel is confirmed
+	// on-chain. Our expectation is that once the channel is confirmed and
+	// Alice reconnects, the channel will show as active on Alice's side.
+	alice.RPC.DisconnectPeer(bob.PubKeyStr)
+
+	// Mine six blocks to confirm the channel funding transaction.
+	ht.MineBlocks(6)
+
+	// Since Alice is still disconnected we expect the channel to still be
+	// deactivated.
+	resp := alice.RPC.ListChannels(&lnrpc.ListChannelsRequest{
+		InactiveOnly: true,
+	})
+	require.Len(ht, resp.Channels, 1)
+
+	// Now Alice reconnects to Bob.
+	alice.RPC.ConnectPeer(&lnrpc.ConnectPeerRequest{
```

Comment:
same here, can use `ht.EnsureConnected(alice, bob` instead to assert they are connected.